src/lib/customclasspath/CustomClasspathRegistrar.ts

```diff
@@ -0,0 +1,101 @@
+import { RegistryJsonHandler } from "./RegistryJsonHandler"
+
+/**
+ * Handles registering classpaths for custom rules and provides visibility into the registry
+ * 
+ * TODO: How can I read the JSON file just once and reuse it everywhere?
+ * 
+ * TODO: verify validity of path. Better to fail now than later while creating catalog.
+ */
+
+
+export enum Engine {
+    PMD = "pmd"
+}
+
+
+export class CustomClasspathRegistrar {
+
+    jsonHandler: RegistryJsonHandler;
+
+    constructor() {
+        this.jsonHandler = new RegistryJsonHandler();
+    }
+
+    public async createEntries(language: string, pathsInput: string[]): Promise<void> {
+
+        const engine = this.identifyEngine();
+        language = language.toLowerCase();
+
+        // Fetch current entries in custom rule register as a Map
+        const engineToLanguageMap = await this.jsonHandler.readCurrentEntries();
+    
+        this.createEngineToLanguageEntry(engineToLanguageMap, engine, language, pathsInput);
+        
+        // Write updated Map to file
+        await this.jsonHandler.updateEntries(engineToLanguageMap);
+
+    }
+
+    public async getEntriesForEngine(engine: Engine): Promise<Map<string, string[]>> {
+        const engineToLanguageMap = await this.jsonHandler.readCurrentEntries();
+        if (engineToLanguageMap.has(engine)) {
+            return engineToLanguageMap.get(engine);
+        }
+
+        // If engine is not found, return an empty Map rather than a null
+        return new Map<string, string[]>();
+    }
+
+    public async getJsonBasedEntriesForEngine(engine: Engine): Promise<string> {
+        const languageToPathMap = await this.getEntriesForEngine(engine);
+        const jsonObject = this.jsonHandler.getJsonOfLanguageMap(languageToPathMap);
+        return JSON.stringify(jsonObject);
+    }
+
+    private createEngineToLanguageEntry(
```

Comment:
The use of 'to' in this name feels weird. Could you maybe rename it something like `createLanguageEntryForEngine`? Also, if you switched to using a JSON instead of a Map, this code could become much simpler.